# Distribución Poisson en Python

Este cuaderno está pensado para usarse en **Google Colab**. Aquí encontrarás:

- qué es la distribución Poisson,
- cómo calcular probabilidades como $P(X=k)$, $P(X\le k)$ y $P(X\ge k)$,
- cómo hallar la **media** y la **varianza**,
- cómo simular datos,
- cómo construir un **histograma**,
- y explicaciones paso a paso de lo que hace cada bloque de código.


In [1]:
import numpy as np
import matplotlib.pyplot as plt
from scipy.stats import poisson


## 1. Idea básica
La distribución Poisson modela el número de eventos en un intervalo fijo. Si $X \sim \text{Poisson}(\lambda)$, entonces:
$$P(X=k)=\frac{e^{-\lambda}\lambda^k}{k!}, \qquad k=0,1,2,\dots$$


In [2]:
lam = 3
print('Lambda (promedio esperado) =', lam)

Lambda (promedio esperado) = 3


## 2. Cálculo de probabilidades

In [3]:
k = 5
print(f'P(X = {k}) = {poisson.pmf(k, lam):.6f}')
print(f'P(X <= {k}) = {poisson.cdf(k, lam):.6f}')
print(f'P(X >= {k}) = {poisson.sf(k - 1, lam):.6f}')

P(X = 5) = 0.100819
P(X <= 5) = 0.916082
P(X >= 5) = 0.184737


In [4]:
datos = np.array([0, 1, 2, 3, 2, 1, 0, 2, 3, 4, 2, 1, 2, 3, 2])
media_m = np.mean(datos)
varianza_m = np.var(datos, ddof=1)

print(f"Media muestral: {media_m:.4f}")
print(f"Varianza muestral: {varianza_m:.4f}")

Media muestral: 1.8667
Varianza muestral: 1.2667


In [5]:
prob_2 = poisson.pmf(2, media_m)
print(f"P(X = 2) con lambda={media_m:.2f}: {prob_2:.4f}")

P(X = 2) con lambda=1.87: 0.2694


### c) Análisis de Causa Raíz (Bayes y Poisson)

Existen dos tipos de servidores con diferentes tasas de fallo:
- **Servidor A:** $\lambda_A = 2$ (65% de las horas).
- **Servidor B:** $\lambda_B = 1$ (35% de las horas).

Si observamos exactamente 2 fallos, ¿cuál es la probabilidad de que provengan del Servidor A?

In [6]:
p_a = 0.65
p_b = 0.35
p_2_dado_a = poisson.pmf(2, 2)
p_2_dado_b = poisson.pmf(2, 1)

p_total_2 = (p_2_dado_a * p_a) + (p_2_dado_b * p_b)
p_a_dado_2 = (p_2_dado_a * p_a) / p_total_2

print(f"Probabilidad de que sea el Servidor A: {p_a_dado_2:.4f}")

Probabilidad de que sea el Servidor A: 0.7321


**Interpretación:** Dado que el servidor A tiene una tasa de fallos más alta y es más frecuente en el tiempo de observación, hay una probabilidad del 73.2% de que una hora con 2 fallos corresponda a dicho servidor.